File: python/python_bandwidth_bxml/verbs/base_verb.py
"""
abstract_bxml_verb.py

Defines the abstract class for all BXML verbs

@author Jacob Mulford
@copyright Bandwidth INC
"""

from abc import ABC, abstractmethod

class AbstractBxmlVerb(ABC):

    def __init__(self, values):
        """
        Initializes the values attribute to build the BXML

        @param values [dict]: The

File: nodejs/lib/verbs/baseVerb.js
/**
 * baseVerb.js
 *
 * Represents the base abstract class for BXML verbs
 *
 * @author Jacob Mulford
 * @copyright Bandwidth INC
 */

class BaseVerb {

    /**
     * Converts the verb class into its object representation for
     * BXML conversion. To be overwritten
     * by all verb classes
     *
     * @param {XMLDocument} xmlDocument The xml document currently being built
     * @return {XMLDocument} The xml document with the verb added to it
     */
    addXml(xmlDocument) {
        return xmlDocument;
    }
}

module.exports = BaseVerb;
